src/Input/Silk.NET.Input.Common/Internals/ReadOnlyCollectionListAdapter.cs

```diff
@@ -0,0 +1,26 @@
+﻿// This file is part of Silk.NET.
+// 
+// You may modify and distribute Silk.NET under the terms
+// of the MIT license. See the LICENSE file for details.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Silk.NET.Input.Internals
+{
+    internal struct ReadOnlyCollectionListAdapter<T> : IReadOnlyList<T>
+    {
+        private readonly IReadOnlyCollection<T> _col;
+
+        public ReadOnlyCollectionListAdapter(IReadOnlyCollection<T> col)
+        {
+            _col = col;
+        }
+
+        public IEnumerator<T> GetEnumerator() => _col.GetEnumerator();
+        IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+        public int Count => _col.Count;
+        public T this[int index] => _col.ElementAt(index);
+    }
+}
```

Comment:
Bit of a weird helper, ElementAt may (or may not) be very expensive, so I don't think this is a good idea.

Reply:
ElementAt isn't that expensive, it essentially just calls MoveNext `index` number of times which is fine for the sizes we're dealing with here.